src/luminator.ts

```diff
@@ -0,0 +1,160 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+import * as dns from 'dns';
+import * as HttpsProxyAgent from 'https-proxy-agent';
+
+type Proxy = {
+  host: string;
+  port: number;
+  auth: {
+    username: string;
+    password: string;
+  },
+}
+
+interface IProxyManagerOption {
+  host: string;
+  port: number;
+  auth: string;
+}
+
+/**
+ * luminator doc
+ */
+class Luminator {
+  public static STATUS_CODE_FOR_RETRY: number[] = [403, 429, 502, 503];
+  private static readonly USER_AGENT: string =
+    'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.36';
+  private static readonly SWITCH_IP_EVERY_N_REQ: number = 30;
+  private static readonly MAX_FAILURES: number = 3;
+  private static readonly REQ_TIMEOUT: number = 60 * 1000;
+  private static readonly MAX_FAILURES_REQ: number = 11;
+  private failuresCountReq: number = 0;
+  private nReqForExitNode: number = 0;
+  private failCount: number = 0;
+  private readonly username: string;
+  private readonly password: string;
+  private readonly superProxy: string;
+  private readonly country: string;
+  private readonly port: number;
+
+  private superProxyHost: string;
+  private superProxyUrl: Proxy;
+  private sessionId: number;
+
+  constructor(username: string, password: string, superProxy: string = 'NL', country: string = 'fr', port: number = 22225) {
+    this.username = username;
+    this.password = password;
+    this.superProxy = superProxy;
+    this.country = country;
+    this.port = port;
+    this.switchSessionId();
+  }
+
+  private static statusCodeRequiresExitNodeSwitch(statusCode: number): boolean {
```

Comment:
- JSdoc is missing
- method name to renamed (must be more shorter & more explicit)